packages/dds/tree/src/shared-tree-core/branch.ts

```diff
@@ -54,50 +51,12 @@ export type SharedTreeBranchChange<TChange> =
 			removedCommits: readonly [GraphCommit<TChange>, ...GraphCommit<TChange>[]];
 	  }
 	| {
-			type: "replace";
+			type: "rebase";
 			change: TaggedChange<TChange> | undefined;
 			removedCommits: readonly GraphCommit<TChange>[];
 			newCommits: readonly GraphCommit<TChange>[];
```

Comment:
I think this would be a good time to rename/refactor `removedCommits` and `newCommits`. I'd advocate for having...
* `originalCommits`: the sequence of commits that were after lowest common ancestor on the source branch before rebasing
* `newBaseCommits`: the sequence of commits that are/were after lowest common ancestor on the destination branch
* `rebasedCommits`: the sequence of commits that are now after the new base commits.
Before the rebase:
//   (0)─(common ancestors)─(LCA)┯─(new base commits)
//                               └─(original commits)
After the rebase:
//   (0)─(common ancestors)─(LCA)─(new base commits)┯
//                                                  └─(rebased commits)

I think that makes the fields on this object both easier to grasp and more alined with the "rebase" terminology (as opposed to being aligned with the "replace" terminology). Looking at how these fields are used in practice though... well, they're not used much. The only usage could probably be replaced with a call to get the branch head. Sooooo... I'm not sure what's best. Just remove them?

Reply:
If I could just remove them right now, I would. Unfortunately, in the case you're thinking of where we could "get the branch head" we _can't_ just get the head because the code is running in beforeChange, not afterChange, so the head hasn't been updated yet. So we at least need something like newCommits or similar passed through the event. And if so, what else, if anything. I like your suggestion, but as you pointed out its mostly unnecessary except for this one little case. 

Or, maybe we can dodge the case entirely by changing when the forest gets updated. Really, the fact that the Forest gets updated during beforeChange is kind of a hack. It ought to happen in afterChange, just with some other mechanism for preventing the sending of malformed ops. There are other reasons we want to move it to afterChange. For example, our top level public nodeChange and treeChange events fire as a result of this code. So right now a user can't fork a branch (or do branching operations in general) in an event callback because they'll be one commit behind. We're presenting the user a view where the forest is forward in time but the commit graph is backwards in time.

So, I think I'd like to lump all that together into one task, file it, and fix all this stuff at once in a later PR. Is that alright with you?